tests/unit/amo/components/TestSearchFilters.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+
+import SearchFilters, {
+  SearchFiltersBase,
+} from 'amo/components/SearchFilters';
+import { searchStart } from 'core/actions/search';
+import { ADDON_TYPE_EXTENSION, OS_LINUX } from 'core/constants';
+import { convertFiltersToQueryParams } from 'core/searchUtils';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import {
+  createStubErrorHandler,
+  getFakeI18nInst,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  const fakeEvent = { preventDefault: sinon.stub() };
+  let context;
+
+  function render({
+    filters = {},
+    pathname = '/search/',
+    ...props
+  } = {}) {
+    const errorHandler = createStubErrorHandler();
+    const { store } = dispatchClientMetadata();
+
+    store.dispatch(searchStart({ errorHandlerId: errorHandler.id, filters }));
+
+    return shallowUntilTarget(
+      <SearchFilters
+        i18n={getFakeI18nInst()}
+        store={store}
+        pathname={pathname}
+        {...props}
+      />, SearchFiltersBase,
+      { shallowOptions: { context } }
+    );
+  }
+
+  beforeEach(() => {
+    context = { router: { push: sinon.stub() } };
```

Comment:
I don't think you need this. You should be able to access the router simply as `props.router`, no?

Reply:
Nope, actually, that won't work when tested. We do it this way elsewhere and using `withRouter` won't work well unless we have at least react-router 3. Also `withRouter` is really just a convenience anyway; [using `context` is still okay](https://github.com/ReactTraining/react-router/issues/3350#issuecomment-211577962).

Reply:
I just realised, after looking at another patch, I was using `withRouter` wrong. This can use the `withRouter` HOC; I'll move it over.